### PR TITLE
Add more title options

### PR DESCRIPTION
**Pull request by KevinBatdorf**
This PR adds a couple more ways to discover a component's title.

Some of these might not fit given they are not standard, but I feel they offer flexibility. I can remove them if no one agrees.

`x-title` - Let's the user override the title and not have it affect anything else.
`x-id` - Used in the `$component` helper so if it exists, the user has intended it to be an identifier
`aria-label` - Accessibility label so better than `role`

This also pushes `role` below `x-data` (if a function name) as `role` is usually pretty generic, while a function name is often a better identifier. Objections?

I also added a unit test to check the naming. Visual test passed too. 

Not sure how to test the Livewire portion though, or if we event want to.

### CODE REVIEW

**Comment by KevinBatdorf** [comment]
First PR so wanted to keep it light :D

File: packages/shell-chrome/src/utils.js [hunk 1]
```diff
@@ -138,3 +138,35 @@ export function flattenSingleAttribute(
...
+
+            if (wire.__instance) {
+                return 'livewire:' + wire.__instance.fingerprint.name
+            }
+        } catch (e) {}
+    }
+}
+
+function convertFunctionName(functionName) {
+    if (functionName.indexOf('{') === 0) return
```
Comment (HugoDF) [inline packages/shell-chrome/src/utils.js:170]: bit easier to read
```suggestion
    if (functionName.startsWith('{')) return
```

File: packages/shell-chrome/src/utils.js [hunk 2]
```diff
@@ -138,3 +138,35 @@ export function flattenSingleAttribute(
...
+            const wire = window.livewire.find(wireId)
+
+            if (wire.__instance) {
+                return 'livewire:' + wire.__instance.fingerprint.name
+            }
+        } catch (e) {}
+    }
+}
+
+function convertFunctionName(functionName) {
```
Comment (HugoDF) [inline packages/shell-chrome/src/utils.js:169]:
```suggestion
function extractFunctionName(xDataExpression) {
```

File: packages/shell-chrome/src/utils.js [hunk 3]
```diff
@@ -138,3 +138,35 @@ export function flattenSingleAttribute(
...
+    if (functionName.indexOf('{') === 0) return
+    return functionName.replace(/\(([^\)]+)\)/, '').replace('()', '')
```
Comment (HugoDF) [inline packages/shell-chrome/src/utils.js:171]: 👀 I have no idea what that does 😄  worth leaving a comment to help visualise what that first replace does?

File: tests/utils.spec.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import Alpine from 'alpinejs'
+import { getComponentName, convertFunctionName } from '../packages/shell-chrome/src/utils'
+import { waitFor } from '@testing-library/dom'
+
+test('getComponentName > will try mutltiple scenarios to determine component name', async () => {
```
Comment (HugoDF) [inline tests/utils.spec.js:5]:
```suggestion
test('getComponentName > will try multiple scenarios to determine component name', async () => {
```

File: tests/utils.spec.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import Alpine from 'alpinejs'
+import { getComponentName, convertFunctionName } from '../packages/shell-chrome/src/utils'
```
Comment (HugoDF) [inline tests/utils.spec.js:2]: unused imports?
```suggestion
import { getComponentName } from '../packages/shell-chrome/src/utils'
```

**Review by HugoDF — COMMENTED** [review]
Happy to merge this.

Couple of comments & since we're adding tests again, it would be good to run them on CI, have a look at https://github.com/alpine-collective/alpinejs-devtools/blob/master/.github/workflows/build.yml#L27

**Comment by HugoDF** [comment]
Also remembered that we've got a note about this in the README that we should probably update.

File: packages/shell-chrome/src/utils.js (hunk 2 above)
Comment (KevinBatdorf) [reply packages/shell-chrome/src/utils.js]: Yeah better. In Inline DevTools I actually convert it to something like `myFunction()` to `My Function`  with the space and all, but felt maybe that wouldn't be as useful here.

File: packages/shell-chrome/src/utils.js (hunk 3 above)
Comment (KevinBatdorf) [reply packages/shell-chrome/src/utils.js]: That actually handles parameters like `myFunction(param)`. I added a note

**Review by HugoDF — APPROVED** [review]

**Pull request merged**